src/check.h.in

```diff
@@ -659,6 +660,117 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
  */
 #define ck_assert_uint_ge(X, Y) _ck_assert_uint(X, >=, Y)
 
+/* Floating point number comparison macros with improved output
+ * compared to ck_assert(). */
+/* OP may be any comparison operator. */
+#define _ck_assert_floating(X, OP, Y) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  ck_assert_msg(_ck_x OP _ck_y, "Assertion '%s' failed: %s == %.6Lf, %s == %.6Lf", #X" "#OP" "#Y, #X, _ck_x, #Y, _ck_y); \
```

Comment:
I thought about it. But tests should give human readable output. So in case of 0.001 and 1.0e-3, 0.001 is clear. But 1.0e-3 spends your time to do some calculations in your mind. In this case %Lg is universal solution.  Just didn't checked what output could be on different compilers and system. In any case I'll make some changes in a week or two to tests making use regular expressions for output analysis (if don't mind) through nan and -nan problem appeared, so no problem, I'll add %Lg everywhere. I can contribute to open source only in free from work time so for now it wouldn't be fast.